stan/math/gpu/kernels/lower_tri_inverse_step2.hpp

```diff
@@ -0,0 +1,122 @@
+#ifndef STAN_MATH_GPU_KERNELS_LOWER_TRI_INVERSE_HPP
+#define STAN_MATH_GPU_KERNELS_LOWER_TRI_INVERSE_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/gpu/kernel_cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+// \cond
+const char* lower_tri_inverse_step2_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * Calculates the intermediate products in the lower_tri_inverse
+     *
+     * @param[in] A input matrix that is being inverted.
+     * @param[out] C temporary matrix to store the intermediate results.
+     * @param M The number of rows for A.
```

Comment:
We use the same conventions as in the matrix multiplication https://github.com/stan-dev/math/blob/develop/stan/math/gpu/kernels/matrix_multiply.hpp
because this is actually just a special case of matrix multiply. 

M, N, K are used everywhere in the GPU literature when doing matrix multiplication examples and we then also used it here, I guess a force of habit. But I do agree that a more descriptive name would be better.

Reply:
Hmm, I see. I suppose it's probably better to follow the existing literature. Maybe we can link to a cheat sheet or something in some higher level GPU doc.